File: api/features/item/item_service.py
from fastapi import Depends
from models.item import Item
from features.item.item_repository import ItemRepository


class ItemService:
    def __init__(self, repo: ItemRepository = Depends()):
        self.repo = repo

    def get_items(self):
        return self.repo.get_all()

    def add_item(self, item: Item):
        self.repo.add_item(item)
        return item 

    def update_item(self, item: Item):
        return self.repo.update_item(item)

    def delete_item(self, item_id: str):
        return self.repo.delete_item(item_id)

File: client/src/features/items/services/itemService.ts
// import { camel_to_snake } from "../../utils/apiMapper";
import { axiosClient } from "../../utils/axiosClient";
import { Item } from "../types/Item";

export const itemService = {
  async getItems(): Promise<Item[]> {
    const res = await axiosClient.get("/api/items");
    return res.data;
  },
  async addItem(item: Item): Promise<void> {
    await axiosClient.post("/api/items", item);
  },
  async updateItem(item: Item): Promise<void> {
    await axiosClient.put(`/api/items/${item.id}`, item);
  },
  async deleteItem(id: string): Promise<void> {
    await axiosClient.delete(`/api/items/${id}`);
  },
};
